crates/git/src/hosting_provider.rs

```diff
@@ -71,24 +71,25 @@ impl HostingProvider {
         commit: Oid,
         client: Arc<dyn HttpClient>,
     ) -> Result<Option<Url>> {
-        match self {
+        let author = match self {
             HostingProvider::Github => {
                 let commit = commit.to_string();
+                github::fetch_github_commit_author(repo_owner, repo, &commit, &client).await?
+            }
+            HostingProvider::Codeberg => {
+                let commit = commit.to_string();
+                codeberg::fetch_codeberg_commit_author(repo_owner, repo, &commit, &client).await?
+            }
+            _ => None,
+        };
 
-                let author =
-                    github::fetch_github_commit_author(repo_owner, repo, &commit, &client).await?;
-
-                let url = if let Some(author) = author {
-                    let mut url = Url::parse(&author.avatar_url)?;
-                    url.set_query(Some("size=128"));
-                    Some(url)
-                } else {
-                    None
-                };
+        Ok(author
+            .map(|author| -> Result<Url, url::ParseError> {
+                let mut url = Url::parse(&author.avatar_url)?;
+                url.set_query(Some("size=128"));
```

Comment:
Does codeberg even support this?

Reply:
Size? Probably not, but it doesn't break it, so whatever I guess?

Reply:
I can add a match inside to only set it for GitHub